jupyter_notebooks/Book_1_Intro_Williamson_2.ipynb

```diff
@@ -0,0 +1,380 @@
+    "                          dumpfreq=dumpfreq,\n",
+    "                          steady_state_error_fields=['u', 'D'],\n",
+    "                          log_level='INFO')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1e1ae651",
+   "metadata": {},
+   "source": [
+    "We can specify which diagnostics we wish to record over a simulation. The list of avaliable diagnostics can be found in the gusto source code: https://github.com/firedrakeproject/gusto/blob/main/gusto/diagnostics.py \\\n",
+    "We pass these diagnostics into the State function. This will initialise all the necessary components of the test case. Gusto will update this state after each iteration of the chosen time-stepper."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "db9d61a0",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "gusto:INFO Physical parameters that take non-default values:\n",
+      "gusto:INFO H: 5960.0\n"
+     ]
+    }
+   ],
+   "source": [
+    "diagnostic_fields = [RelativeVorticity(), PotentialVorticity(),\n",
+    "                         ShallowWaterKineticEnergy(),\n",
+    "                         ShallowWaterPotentialEnergy(),\n",
+    "                         ShallowWaterPotentialEnstrophy(),\n",
+    "                        CourantNumber()]\n",
+    "\n",
+    "state = State(mesh,\n",
+    "              dt=dt,\n",
+    "              output=output,\n",
+    "              parameters=parameters,\n",
+    "              diagnostic_fields=diagnostic_fields)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "80f8c00a",
+   "metadata": {},
+   "source": [
+    "A spatially varying Coriolis force is defined over the sphere. We then can set-up our form of the shallow water equations by passing the state and Coriolis parameter into the ShallowWaterEquations() function. We additionally specify the function space (BDM) and degree (1) that we want to use with the Compatiable Finite Element method. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "8ed9914e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Create a spatially varying function for the Coriolis force:\n",
+    "Omega = parameters.Omega\n",
+    "x = SpatialCoordinate(mesh)\n",
+    "fexpr = 2*Omega*x[2]/R\n",
+    "eqns = ShallowWaterEquations(state, \"BDM\", 1, fexpr=fexpr)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9918a45e",
+   "metadata": {},
+   "source": [
+    "We are now ready to specify the initial conditions.\n",
+    "Due to our choice of function spaces for the velocity and depth, the intialisations of each variable use projection \n",
+    "and interpolation operations respectively. Williamson 2 specifies initial conditions for a 'Global Steady State Non-linear Zonal Geostrophic Zone' (do I need this detail?). \n",
```

Comment:
I think you don't need this detail -- we've explained the initial condition at the top so I think it's OK to leave this out